pyproject.toml

```diff
@@ -8,7 +8,7 @@ readme = "README.md"
 repository = "https://github.com/OpenDevin/OpenDevin"
 
 [tool.poetry.dependencies]
-python = "^3.11"
+python = ">=3.11,<4.0"
```

Comment:
Aren't these two lines equivalent?

Reply:
no for the first line it selected just the 3.11 but for the second it will take 3.11, 3.12, 3.13....4.0 and using `>=3.11, <4.0` instead of >=3.11 is because llama-index support till python 4.0

Reply:
The caret symbol `^` is short hand for >= 3.11 but not 4.
see: https://python-poetry.org/docs/dependency-specification/

Reply:
I didn't know that I thought that it selects the jus the correspondent version, thanks for the info!

Reply:
so I'll basically close this PR